src/typescript-ast-nodes.js

```diff
@@ -0,0 +1,131 @@
+module.exports = function(fork) {
+  fork.use(require("ast-types/def/es7"));
+
+  var types = fork.use(require("ast-types/lib/types"));
+  var def = types.Type.def;
+  var or = types.Type.or;
+  var defaults = fork.use(require("ast-types/lib/shared")).defaults;
+  
+  // Ambient
+  def("TSAmbientVariableDefinition").bases("VariableDeclaration");
+  
+  def("TSInterfaceDeclaration")
+    .bases("Declaration")
+    .build("name", "typeParameters", "members")
+    .field("name", def("Identifier"))
+    .field(
+      "typeParameters",
+      or(def("TypeParameterDeclaration"), null),
+      defaults["null"]
+    )
+    .field("members", [def("TSSignature")]);
+  // .field("body", def("ObjectTypeAnnotation"))
+  // .field("extends", [def("InterfaceExtends")]);
+  
+  def("TSKeyword").bases("Node");
+  
+  def("TSType").bases("Node");
+  
+  def("TypeElement").bases("Node");
+  
+  def("TSSignature")
+    .bases("TypeElement")
+    .build("typeParameters", "parameters", "typeAnnotation")
+    .field(
+      "typeParameters",
+      or(def("TypeParameterDeclaration"), null),
+      defaults["null"]
+    )
+    .field("parameters", [def("Identifier")])
+    .field("typeAnnotation", def("TSType"));
+  
+  def("TSAnyKeyword").bases("TSKeyword");
+  
+  def("TSBooleanKeyword").bases("TSKeyword");
+  
+  def("TSNeverKeyword").bases("TSKeyword");
+  
+  def("TSNumberKeyword").bases("TSKeyword");
+  
+  def("TSObjectKeyword").bases("TSKeyword");
+  
+  def("TSReadonlyKeyword").bases("TSKeyword");
```

Comment:
I think the file is for the actual TypeScript AST not the converted one from https://github.com/eslint/typescript-eslint-parser